partition/public.go

```diff
@@ -19,17 +19,13 @@
 
 package partition
 
-import (
-	"github.com/ubuntu-core/snappy/helpers"
-)
-
 // BootloaderDir returns the full path to the (mounted and writable)
 // bootloader-specific boot directory.
 func BootloaderDir() string {
-	if helpers.FileExists(bootloaderUbootDir()) {
-		return bootloaderUbootDir()
-	} else if helpers.FileExists(bootloaderGrubDir()) {
-		return bootloaderGrubDir()
+	if u := newUboot(); u != nil {
+		return u.Dir()
+	} else if g := newGrub(); g != nil {
+		return g.Dir()
 	}
 
 	return ""
```

Comment:
This should probably return an error, as a hint to the call site that it can actually fail. It also feels like a slightly curious construct.. what use is it to have a bootloader dir without knowing that the bootloader is?  And why does it make sense to have it public, but not the rest of the API?  Honest questions. The only use case I've seen of BootloaderDir is in a piece of code saying it's terrible. :-)  Can we get rid of it?